precompiles/ArbOwner.go

```diff
@@ -157,3 +161,50 @@ func (con ArbOwner) ReleaseL1PricerSurplusFunds(c ctx, evm mech, maxWeiToRelease
 	}
 	return weiToTransfer, nil
 }
+
+func (con ArbOwner) SetChainConfig(c ctx, evm mech, serializedChainConfig []byte) error {
+	if c == nil {
+		return errors.New("nil context")
+	}
+	if c.txProcessor == nil {
+		return errors.New("uninitialized tx processor")
+	}
+	if c.txProcessor.MsgIsGasEstimation() {
```

Comment:
Sorry to recommend this change after you already wrote the code, but could we change this to `MsgIsNonMutating()`? We'd want that method to return true if it's either gas estimation mode or eth_call mode. I'm realizing that the governance process might not rely on gas estimation until the upgrade is already underway, so erroring if e.g. the governance seatbelt checks this with an `eth_call` is also important.